modules/integration/tests-integration/tests-amqp/src/test/java/org/wso2/mb/integration/tests/amqp/functional/dtx/DtxStartPositiveTestCase.java

```
@@ -311,6 +311,106 @@ public void xaStartJoinQueuePublishTestCase()
+        producerTwo.send(sessionOne.createTextMessage("Test 2"));
+
+        xaResourceOne.end(xid, XAResource.TMSUCCESS);
+
+        // subscribe and see if the message is received
+        ConnectionFactory nonXaConnectionFactory = (ConnectionFactory) initialContext
+                .lookup(JMSClientHelper.QUEUE_CONNECTION_FACTORY);
+        Connection nonXaQueueConnection = nonXaConnectionFactory.createConnection();
+        nonXaQueueConnection.start();
+        Session nonXaQueueSession = nonXaQueueConnection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+        MessageConsumer messageConsumer = nonXaQueueSession.createConsumer(xaTestQueue);
+
+        // wait 5 seconds
+        Message receive = messageConsumer.receive(5000);
+        Assert.assertNull(receive, "Message received before committing");
+
+        xaResourceOne.prepare(xid);
+        xaResourceOne.commit(xid, false);
+
+        xaConnectionOne.close();
+
+        //This is only added to find out the reason for the intermittent failure of this test method. Should be removed
+        // once the issue is identified.
+        try {
+            // Logging in
+            LoginLogoutClient loginLogoutClientForAdmin = new LoginLogoutClient(super.automationContext);
+            String sessionCookie = loginLogoutClientForAdmin.login();
+            AndesAdminClient admin = new AndesAdminClient(super.backendURL, sessionCookie);
+
+            //Check message count in queue
+            org.wso2.carbon.andes.stub.admin.types.Message[] queueOneMessages
+                    = admin.browseQueue(queueName, 0, 10);
+            Assert.assertEquals(queueOneMessages.length, 2, "Message not published to queue " + queueName);
+
+            //Logging out
+            loginLogoutClientForAdmin.logout();
+
+        } catch (RemoteException | AutomationUtilException | AndesAdminServiceBrokerManagerAdminException
+                | LogoutAuthenticationExceptionException e) {
+            e.printStackTrace();
```

Don't we need to propagate the error forward if an exception occurs?